src/components/Search/FileUploader.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import { Col, Upload, Button } from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+
+
+class FileUploader extends Component {
+
+    beforeUpload = file => {
+        const reader = new FileReader();
+        if (this.validateSize(file)) {
+            reader.onload = () => {
+                const json = JSON.parse(reader.result);
+                const geoid = file.name.replace('geo','').replace('.json','');
+
+                if (this.validateFile(json)) {
+                    this.props.onUpload(json, geoid);
+                }
+            };
+            reader.readAsText(file);
+
+            // Prevent upload
+            return false;
+        }
+    }
+
+    validateFile = (json) => {
+        // TODO: validation step
+        console.log(json)
+    }
+
+    validateSize = (file) => {
+        const size = 1024 * 1024 * 100;  //100mb
+        let error = '';
```

Comment:
since error is only used in the alert, it doesn't need to be init here. You could also just enter the error string directly into the alert.